### PR TITLE
Add mandatory IDP referrer check

### PR DESCRIPTION
**Pull request by npm1**
Fixes https://github.com/fedidcg/FedCM/issues/325

***
<a href="https://pr-preview.s3.amazonaws.com/npm1/FedCM/pull/338.html" title="Last updated on Aug 26, 2022, 5:27 PM UTC (01bc080)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/338/1928c23...npm1:01bc080.html" title="Last updated on Aug 26, 2022, 5:27 PM UTC (01bc080)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -855,6 +855,10 @@ account_id=123&client_id=client1234&nonce=Ct60bD&disclosure_text_shown=true
 ```
 </div>
 
+An [=IDP=] needs to check the referrer to ensure that a malicious [=RP=] does not receive an ID
+token corresponding to another [=RP=]. In other words, the referrer must match the [=client id=]. As
```
Comment (samuelgoto) [inline spec/index.bs:859]:
```suggestion
token corresponding to another [=RP=]. In other words, the referrer must be represented by the [=client id=]. As
```

Referrers are origins, client_ids are arbitrary strings (which aren't origins necessarily), so string matching wouldn't work.

File: spec/index.bs [hunk 2]
```diff
@@ -855,6 +855,10 @@ account_id=123&client_id=client1234&nonce=Ct60bD&disclosure_text_shown=true
 ```
 </div>
 
+An [=IDP=] needs to check the referrer to ensure that a malicious [=RP=] does not receive an ID
```
Comment (samuelgoto) [inline spec/index.bs:858]: Is this a normative conformace requirement to IDPs? If so, is there something like NOTE: that we could use to collect those?

**Review by samuelgoto — APPROVED** [review]
LGTM 

Will merge when comments are addressed

File: spec/index.bs (hunk 2 above)
Comment (npm1) [reply spec/index.bs:858]: Anything within a note is non-normative text. So I chose to not put it inside a note.

**Comment by npm1** [comment]
ptal

File: spec/index.bs (hunk 2 above)
Comment (npm1) [reply spec/index.bs:865]: Added style per offline discussion.

File: spec/index.bs [hunk 3]
```diff
@@ -855,6 +861,13 @@ account_id=123&client_id=client1234&nonce=Ct60bD&disclosure_text_shown=true
 ```
 </div>
 
+<div class=idp-normative-text>
+An [=IDP=] needs to check the referrer to ensure that a malicious [=RP=] does not receive an ID
```
Comment (kdenhartog) [inline spec/index.bs:865]: Nit: using RFC 2119 language here to identify this more clearly as a normative statement would be helpful.

E.g. An IDP MUST check the referrer header matches the expected client id for the RP. If the referrer does not match the client id the IDP MUST reject the request.

**Review by kdenhartog — COMMENTED** [review]
Editorial nit added

**Pull request merged**